src/security/srv_acl.c

```diff
@@ -473,34 +525,138 @@ ds_sec_check_pool_access(struct daos_acl *acl, struct pool_owner *ownership,
 		return rc;
 	}
 
-	rc = get_auth_sys_payload(token, &authsys);
+	rc = get_sec_capas_for_token(token, ownership, acl,
+				     pool_capas_from_perms, capas);
+	if (rc == 0)
+		filter_pool_capas_based_on_flags(flags, capas);
+
 	auth__token__free_unpacked(token, NULL);
-	if (rc != 0)
-		return rc;
+	return rc;
+}
+
+static uint64_t
+cont_capas_from_perms(uint64_t perms)
+{
+	uint64_t capas = 0;
+
+	if (perms & DAOS_ACL_PERM_READ)
+		capas |= CONT_CAPA_READ_DATA;
+	if (perms & DAOS_ACL_PERM_WRITE)
+		capas |= CONT_CAPA_WRITE_DATA;
+	if (perms & DAOS_ACL_PERM_GET_PROP)
+		capas |= CONT_CAPA_GET_PROP;
+	if (perms & DAOS_ACL_PERM_SET_PROP)
+		capas |= CONT_CAPA_SET_PROP;
+	if (perms & DAOS_ACL_PERM_GET_ACL)
+		capas |= CONT_CAPA_GET_ACL;
+	if (perms & DAOS_ACL_PERM_SET_ACL)
+		capas |= CONT_CAPA_SET_ACL;
+	if (perms & DAOS_ACL_PERM_SET_OWNER)
+		capas |= CONT_CAPA_SET_OWNER;
+
+	return capas;
+}
+
+static void
+filter_cont_capas_based_on_flags(uint64_t flags, uint64_t *capas)
+{
+	if (flags & DAOS_COO_RO)
+		*capas &= CONT_CAPAS_RO_MASK;
+	else if (!(*capas & CONT_CAPAS_RO_MASK) ||
+	         !(*capas & ~CONT_CAPAS_RO_MASK))
```

Comment:
(style)  code indent should use tabs where possible

Reply:
Fixed